# roi_lanes.yaml
# 차선 ROI 소스 경로와 적용 정책 설정.

roi_source:
  format: json
  path: "$(find autocal)/src/nodes/lane_polys.json"
  # lane_polys.json은 ROI 편집 시 자동 갱신됨.
  # 예시: { "IN1": [[u,v], ...], "IN2": [...], "OUT1": [...], "OUT2": [...] }

lane_definition:
  lane_names: ["IN1", "IN2", "IN3", "OUT1", "OUT2", "OUT3"]
  lane_state_keys:
    IN1: "in1"
    IN2: "in2"
    IN3: "in3"
    OUT1: "out1"
    OUT2: "out2"
    OUT3: "out3"

policy:
  anchor_point: bbox_bottom_center     # 옵션: bbox_bottom_center / bbox_center
  min_bbox_area: 1500                  # 최소 bbox 면적 필터

selection:
  mode: nearest_by_radar_range         # 옵션: nearest_by_radar_range / largest_bbox / closest_to_center
  publish_only_one: true

debug:
  publish_roi_debug: true
  roi_debug_topic: /autocal/lane_roi_debug
